File: Design Patterns/Decorator Pattern/Java/com.faculdadevincit.designpatterns.decoratorpattern/src/CafeAdicionalDeLeite.java
public class CafeAdicionalDeLeite extends CafeDecorator{

    public CafeAdicionalDeLeite(ICafe cafe) {
            super(cafe);
        }
    
    @Override
    public double getCusto() {
        double custo = super.getCusto();
        custo += 1.00;
        return custo;
    }

    @Override
    public String getDescricao() {
        String descricao = super.getDescricao();
        descricao += " com leite";
        return descricao;
    }

}

File: Design Patterns/Decorator Pattern/CSharp/FaculdadeVincit.DesingPatterns.DecoratorPattern/CafeAdicionalDeLeite.cs
namespace FaculdadeVincit.DesingPatterns.DecoratorPattern;

public class CafeAdicionalDeLeite : CafeDecorator
{
    public CafeAdicionalDeLeite(ICafe cafe) : base(cafe)
    {
    }

    public override string Descricao => $"{base.Descricao} com leite";

    public override double Custo => base.Custo + 1.00;


}
